src/jit/lower.cpp

```diff
@@ -3841,46 +3841,266 @@ GenTree* Lowering::LowerAdd(GenTree* node)
 }
 
 //------------------------------------------------------------------------
-// LowerUnsignedDivOrMod: transform GT_UDIV/GT_UMOD nodes with a const power of 2
-// divisor into GT_RSZ/GT_AND nodes.
+// GetUnsignedMagicNumberForDivide: Generates a magic number and shift amount for
+// the magic number unsigned division optimization.
 //
 // Arguments:
-//    node - pointer to the GT_UDIV/GT_UMOD node to be lowered
+//    denom - The denominator
+//    shift - Pointer to the shift value to be returned
+//    add   - Pointer to a flag indicating the kind of code to generate
 //
-void Lowering::LowerUnsignedDivOrMod(GenTree* node)
+// Returns:
+//    The magic number.
+//
+// Notes:
+//    This code is previously from UTC where it notes it was taken from
+//   _The_PowerPC_Compiler_Writer's_Guide_, pages 57-58. The paper is is based on
+//   is "Division by invariant integers using multiplication" by Torbjorn Granlund
+//   and Peter L. Montgomery in PLDI 94
+
+template <typename U>
+U GetUnsignedMagicNumberForDivide(U denom, int* shift /*out*/, bool* add /*out*/)
```

Comment:
There's only one occurrence in corelib and a couple of others in System.Runtime.Numerics and System.Xml.ReaderWriter so there's not much to measure. 

FWIW I measured how much time this function takes for `denom` = 10 and it's ~12ns on my machine. The time seems to reach ~70ns for larger denominators.

The book from which this code was adapted actually suggests the possibility of using a precomputed table but then the book was written in 1996, computers where kind of slow back then :)